### DISCUSSION TITLE
Unreachable code warning is not emitted if panic is called as a funcion

### DISCUSSION THREAD

**Discussion by malbarbo** [gleam-lang/gleam#3973] [Ideas & suggestions]
For this code

```gleam
fn f(a) { a }

pub fn main() {
  f(panic())
}
```

the compiler emits the warning "Panic used as a function" but not "Unreachable code" for f.

**Comment by lpil** [comment]
Hello! Do you not think the existing warnings highlights that the code doesn't work?

**Comment by malbarbo** [comment]
The warnings are fine. The issue is that, once the warning "Panic used as a function" is resolved by removing `()` from `panic`, a new warning is emitted: "Unreachable code" for f. Shouldn't both warnings be emitted upfront?

**Comment by lpil** [comment]
I don't mind either way really!

**Comment by jpyamamoto** [comment]
I've been interested in Gleam for a while, and taking advantage of the currently running AoC, I decided to try the language and maybe contribute back. (Shout out to @lpil, thank you for the language).

Looking for a good first contribution, I was gonna try to send a PR fixing this scenario, but after pondering on it I came to the conclusion that maybe it wasn't a great idea. If I may add to the discussion, I don't think it's ideal to assume we know better than the programmer the intention behind the program.

Telling the programmer that `panic` cannot be used as a function when we find `panic()` is reasonable, since `panic` is a reserved word and it is safe to say the programmer is misusing the keyword. But assuming the error produced by `panic()` will be fixed by the programmer writing `panic` and throwing an "Unreachable code" error in advance, is a little too much. That is not the only possible typo, nor the only possible fix for the erroneous code.

I can see a scenario when the programmer actually meant to write `panic_foo()` (or any other valid function name containing the substring `panic` in its name) and getting the "Unreachable error" in advance would be a mistake. The "Unreachable code" error should be thrown only after absolute certainty that the syntactically correct code won't be reached.

Just adding my two cents here. If the community decides that both errors should be thrown in advance, I'd be glad to contribute with a PR for this.

**Comment by lpil** [comment]
Sorry, I'm not sure what you're suggesting we do here. If there a new error message you'd like to propose? Thanks

**Comment by jpyamamoto** [comment]
I'm suggesting we leave the current behavior as currently is, not displaying both warnings in advance.
